LanguageFeatures/Extension-types/static_analysis_extension_types_A08_t06.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a type T is incompatible with await if at least one
+/// of the following criteria holds:
+/// - T is an extension type that does not implement Future.
+/// - T is S?, and S is incompatible with await.
+/// - T is X & B, and either:
+/// - B is incompatible with await, or
+/// - B does not derive a future type, and X is incompatible with await.
+/// - T is a type variable with bound S, and S is incompatible with await.
+/// Consider an expression of the form await e. A compile-time error occurs if
+/// the static type of e is incompatible with await.
+///
+/// @description Checks that it is a compile-time error if `await e` occurs, and
+/// the static type of `e` is `X & B`, and `B` is incompatible with await.
+/// @author sgrekhov22@gmail.com
+/// @issue 54648, 54649
+
+// SharedOptions=--enable-experiment=inline-class
+
+import 'dart:async';
+
+extension type NumET(num _) {}
+extension type IntET(int _) {}
+
+test1<X extends FutureOr<NumET>>(X x) async {
+  if (x is int) {
+    await x; // No error here, int is compatible with await
+  }
+  if (x is IntET) {
```

Comment:
Same issue here, there is no promotion of `x`.

Reply:
Similar issues in test2.

Reply:
Fixed

Reply:
Thanks!